hypothesis-python/src/_hypothesis_pytestplugin.py

```diff
@@ -316,30 +325,13 @@ def pytest_runtest_makereport(item, call):
                 ]
 
     def pytest_terminal_summary(terminalreporter):
-        if not terminalreporter.config.getoption(PRINT_STATISTICS_OPTION):
-            return
-        terminalreporter.section("Hypothesis Statistics")
-
-        def report(properties):
-            for name, value in properties:
-                if name.startswith("hypothesis-statistics-"):
-                    if hasattr(value, "uniobj"):
-                        # Under old versions of pytest, `value` was a `py.xml.raw`
-                        # rather than a string, so we get the (unicode) string off it.
-                        value = value.uniobj
-                    line = base64.b64decode(value.encode()).decode() + "\n\n"
-                    terminalreporter.write_line(line)
-
-        try:
-            global_properties = terminalreporter.config._xml.global_properties
-        except AttributeError:
-            # terminalreporter.stats is a dict, where the empty string appears to
-            # always be the key for a list of _pytest.reports.TestReport objects
-            for test_report in terminalreporter.stats.get("", []):
-                if test_report.when == "teardown":
-                    report(test_report.user_properties)
```

Comment:
I'm not sure what `user_properties`'s primary purpose is, but it's automatically included in the `<testcase>` node of a junit report:
https://github.com/pytest-dev/pytest/blob/00ad12b9db8d3309c361061fe1c4878b2e0cd51c/src/_pytest/junitxml.py#L599-L600

This seems wrong, as `<testcase>` doesn't allow any arbitrary node in xunit2. I think the proper upstream fix should distinguish between report-properties-to-add-to-junit and report-properties-metadata-to-carry-around.

If `user_properties` were distinguished as such, we could use that instead of the semi-hacky `report.__dict__` storing.